openedx_learning/core/components/api.py

```diff
@@ -73,14 +73,82 @@ def create_component_version(
     return component_version
 
 
+def create_next_version(
+    component_pk: int,
+    title: str,
+    content_to_replace: dict[str: int],
+    created: datetime,
+    created_by: int | None = None,
+) -> ComponentVersion:
+    """
+    Create a new ComponentVersion based on the most recent version.
+
+    A very common pattern for making a new ComponentVersion is going to be "make
+    it just like the last version, except changing these one or two things".
+    Before calling this, you should create any new contents via the contents
+    API, since ``content_to_replace`` needs RawContent IDs for the values.
+
+    TODO: Have to add learning_downloadable info to this when it comes time to
+          support static asset download.
+    """
+    # This needs to grab the highest version_num for this Publishable Entity.
+    # This will often be the Draft version, but not always. For instance, if
+    # an entity was soft-deleted, the draft would be None, but the version_num
+    # should pick up from the last edited version. Likewise, a Draft might get
+    # reverted to an earlier version, but we want the latest version_num when
+    # creating the next version.
+    component = Component.objects.get(pk=component_pk)
+    last_version = component.versioning.latest
+    if last_version is None:
+        next_version_num = 1
+    else:
+        next_version_num = last_version.version_num + 1
+
+    with atomic():
+        publishable_entity_version = publishing_api.create_publishable_entity_version(
+            entity_id=component_pk,
+            version_num=next_version_num,
+            title=title,
+            created=created,
+            created_by=created_by,
+        )
+        component_version = ComponentVersion.objects.create(
+            publishable_entity_version=publishable_entity_version,
+            component_id=component_pk,
+        )
+        # First copy the new stuff over...
+        for key, raw_content_pk in content_to_replace.items():
+            ComponentVersionRawContent.objects.create(
+                raw_content_id=raw_content_pk,
+                component_version=component_version,
+                key=key,
+                learner_downloadable=False,
+            )
```

Comment:
Yeah. I was first punting just because I was trying to implement only exactly what I needed to get v2 libraries over, but I should just add it. Thank you.